components/action-mgt/org.wso2.carbon.identity.action.execution/src/main/java/org/wso2/carbon/identity/action/execution/ActionExecutorServiceImpl.java

```diff
@@ -85,26 +85,77 @@ public boolean isExecutionEnabled(ActionType actionType) {
         return ActionExecutorConfig.getInstance().isExecutionForActionTypeEnabled(actionType);
     }
 
+    /**
+     * Resolve the actions that need to be executed for the given action types and execute them.
+     *
+     * @param actionType    Action Type.
+     * @param eventContext  The event context of the corresponding flow.
+     * @param tenantDomain  Tenant domain.
+     * @return Action execution status.
+     */
     public ActionExecutionStatus execute(ActionType actionType, Map<String, Object> eventContext, String tenantDomain)
             throws ActionExecutionException {
 
         try {
             List<Action> actions = getActionsByActionType(actionType, tenantDomain);
             validateActions(actions, actionType);
-            ActionExecutionRequest actionRequest = buildActionExecutionRequest(actionType, eventContext);
-            ActionExecutionResponseProcessor actionExecutionResponseProcessor = getResponseProcessor(actionType);
-
-            Action action = actions.get(0); // As of now only one action is allowed.
-            return Optional.ofNullable(action)
-                    .filter(activeAction -> activeAction.getStatus() == Action.Status.ACTIVE)
-                    .map(activeAction -> executeAction(activeAction, actionRequest, eventContext,
-                            actionExecutionResponseProcessor))
-                    .orElse(new ActionExecutionStatus(ActionExecutionStatus.Status.FAILED, eventContext));
+            // As of now only one action is allowed.
+            Action action = actions.get(0);
+            return execute(action, eventContext);
         } catch (ActionExecutionRuntimeException e) {
             // todo: add to diagnostics
             LOG.debug("Skip executing actions for action type: " + actionType.name(), e);
             return new ActionExecutionStatus(ActionExecutionStatus.Status.FAILED, eventContext);
+        }
+    }
+
+    /**
+     * Resolve the actions by given the action id list and execute them.
+     *
+     * @param actionType    Action Type.
+     * @param actionIds     Lis of action Ids of the actions that need to be executed.
+     * @param eventContext  The event context of the corresponding flow.
+     * @param tenantDomain  Tenant domain.
+     * @return Action execution status.
+     */
+    public ActionExecutionStatus execute(ActionType actionType, String[] actionIds, Map<String, Object> eventContext,
+                                         String tenantDomain) throws ActionExecutionException {
 
+        try {
+            // As of now only one action is allowed.
+            if (actionIds.length != 1) {
```

Comment:
Let's do the validation to make sure this array is not null and not empty

Reply:
Basically extract to a validation method similar to https://github.com/wso2/carbon-identity-framework/pull/5902/files#diff-30dbfcba373317b490119e2dbd937558324dc1d02e3cd745febf0d9631b792ebR101

Reply:
Improved this to be done with  new validateActionIdList() methods.